### DISCUSSION TITLE
Consider leveraging React Native Test App

### DISCUSSION THREAD

**Discussion by thymikee** [callstack/react-native-builder-bob#464] [Ideas]
## Describe the feature

RNTA https://github.com/microsoft/react-native-test-app is a project allowing library developers to add examples for various platforms. I wonder if it would make sense to use it by Bob? I don't have any opinions about it right now, just shooting an idea for discussion.

cc @tido64 @kelset – would be great to have your perspective as well :D 

## Motivation

Maybe there's an opportunity to offload some work to RNTA project, which would make Bob easier to maintain.

**Comment by tido64** [comment]
We discussed this briefly on [Discord](https://discord.com/channels/514829729862516747/550602825894330399/999972117191860285). Personally, I would love to have more people use RNTA. If Bob (or any other libraries for that matter) wants to integrate it, I'll try to assist in any way I can.

I also understand that people want/need the ability to customize it further. I'm hoping that adding support for `@expo/config-plugins` will help with that: https://github.com/microsoft/react-native-test-app/pull/1033

**Comment by kelset** [comment]
+1 on this and what Tommy said.

It's something that we think would help a lot of folks, so we are 👍 in supporting this happening :)

**Comment by shamilovtim** [comment]
Hi guys! I recently had to upgrade the bob / create-react-native-app shell of a library I built 3+ years ago with bob. That meant:
1. I had to upgrade all of the build tools and package management (deps versions, node, npm, ruby, yarn, podspec, scripts, cocoapods, ruby, etc.)
2. I had to upgrade 6 react native major versions of the test app

While 1. was fine and took maybe a couple of hours, I found 2. to be an unmaintainable burden given the velocity of changes in the underlying RN code. Once I got to the native files it was a disaster probably no faster than upgrading 1 major version at a time by hand in the example app (in fact I regretted not bumping one major at a time). My strategy was to run a brand new `npx create-react-native-library` and then one by one replace native files. I think RN test app like Expo CNG would've made this a whole lot easier. 

I'm wondering if there are any plans to roll in React Native Test App or Expo CNG? Also curious about compatibility. If I were to go off on my own and implement one of these upon the example app in a standalone `bob` project would it break some of the symlinking going on with the iOS "Development Pod" or the Android development project?

**Comment by kraenhansen** [comment]
I believe this landed with the merge of https://github.com/callstack/react-native-builder-bob/pull/572

**Reply by tido64** [reply]
It was temporarily disabled in #658 pending some work to support codegen. @atlj, what's the latest here? Do we have a timeline for when it gets enabled again?

Edit: Oh, it looks like it was re-enabled again a few days back.

**Reply by atlj** [reply]
Hey, we're currently dealing with lots of bugs related to our codegen implementation. I'd say once they are sorted out we can bring the implementation back. @tido64 The main thing we need to add to our testing app template is the ability to call our custom codegen script when the library maintainer builds their example app.  We have edited the Podfile for iOS and added a gradle script for Android to achieve this with the community CLI example.

**Reply by thymikee** [reply]
How’s this going to bring it back?
